src/core/mount.c

```diff
@@ -1750,9 +1783,54 @@ static int mount_load_proc_self_mountinfo(Manager *m, bool set_flags) {
                 if (!device || !path)
                         continue;
 
+                if (ignore_mount(path, fstype))
+                        continue;
+
+                size = snprintf(mount_entry, mount_entry_size, "%s %s %s %s", device, path, fstype, options);
+
+                if ((size_t)size >= mount_entry_size) {
+                        if (mount_entry)
+                                free(mount_entry);
+                        mount_entry_size = 1024 * (size / 1024 + 1);
+                        mount_entry = new(char, mount_entry_size);
+                        if (!mount_entry) {
+                                mount_entry_size = 0;
+                                return -ENOMEM;
+                        }
+                        size = snprintf(mount_entry, mount_entry_size, "%s %s %s %s", device, path, fstype, options);
+                }
+
+                mu = hashmap_get2(mountinfo_hashmap, mount_entry, (void **)&key);
```

Comment:
Why use a string as the hashmap key? It would be nicer to hash the relevant items into a single number. This will make things more efficient and avoid the need to allocate the key.

Reply:
I don't see how this could optimize anything here. The hash function would basically be "concatenate hash of device + has of path + hash of fstype + hash of options" which all are "small" values, I'm pretty sure hashing a long string is more efficient and quicker.
Additionally the allocation of `mount_entry` only happens once (or twice).

Reply:
Exactly, that's the point. The whole mount line is what makes it unique (from Mount unit point of view).